CommonCompilerConfig.cmake

```diff
@@ -102,6 +104,22 @@ endif()
 message(STATUS "CMAKE BITNESS=${BITNESS}")
 
 
+# Check memory limits (in megabytes)
+if(CMAKE_MAJOR_VERSION GREATER 2)
+  cmake_host_system_information(RESULT total_physical_memory QUERY TOTAL_PHYSICAL_MEMORY)
+  cmake_host_system_information(RESULT total_virtual_memory QUERY TOTAL_VIRTUAL_MEMORY)
+  math(EXPR total_memory "${total_physical_memory}+${total_virtual_memory}")
```

Comment:
I am not sure whether it makes sense to add physical and virtual memory together. Virtual memory will map onto physical memory. Insufficient virtual memory will make it impossible to complete the compilation. Insufficient physical memory will likely make compilation very slow (thrashing to SWAP) if swap is enabled or impossible if swap is disabled in the kernel. Would `AVAILABLE_VIRTUAL_MEMORY` and `AVAILABLE_PHYSICAL_MEMORY` be more useful than the totals?